src/DataModel/IApiDocumentation.ts

```diff
@@ -0,0 +1,9 @@
+import {IMetadataDescription} from "./IMetadataDescription";
+import {IData} from "./IData";
+
+export interface IApiDocumentation extends IMetadataDescription
+{
+    entryPoint: string;
+
+    getEntryPoint(): Promise<IData>;
```

Comment:
As for the Single Responsibility Principle - I don't think it would be a violation - It's a functional inheritance and actual implementation could be a simple wrapper on top of an actual implementation.

Indeed there is an inconsistency in use cases - some places uses method chaining, i.e.:
```javascript
var operation = client.get("http://example.com")
    .getApiDocumentation()
    .getEntryPoint()
```
And some other are using client as an invocation facility for operations, i.e.:
```javascript
var operation = client.get("http://example.com")
    .getApiDocumentation()
    .getEntryPoint()
    .getOperations()
    .filter(op => op.method == "POST" && op.expects == "Event")[0];
client.invoke(operation, event);
```
I'm not sure which one is better. The latter seems cleaner, but it would force client to do quite a lot (i.e. make distinction on how to work with simple Url string, Link, Iri template or an operation).

As for your example, I wouldn't like to see developers create instances - I'd prefer our API to do it on it's own so developers don't need to bother about real implementation.